packages/grafana-ui/src/components/VizLegend/SeriesIcon.tsx

```diff
@@ -1,20 +1,44 @@
 import React, { CSSProperties } from 'react';
+import { useTheme2 } from '../../themes';
+import { fieldColorModeRegistry } from '@grafana/data';
 
 export interface Props extends React.HTMLAttributes<HTMLDivElement> {
-  color: string;
+  color?: string;
+  gradient?: string;
 }
 
-export const SeriesIcon = React.forwardRef<HTMLDivElement, Props>(({ color, className, ...restProps }, ref) => {
-  const styles: CSSProperties = {
-    backgroundColor: color,
-    width: '14px',
-    height: '4px',
-    borderRadius: '1px',
-    display: 'inline-block',
-    marginRight: '8px',
-  };
+export const SeriesIcon = React.forwardRef<HTMLDivElement, Props>(
+  ({ color, className, gradient, ...restProps }, ref) => {
+    const theme = useTheme2();
+    let cssColor: string;
 
-  return <div ref={ref} className={className} style={styles} {...restProps} />;
-});
+    if (gradient) {
+      const colors = fieldColorModeRegistry.get(gradient).getColors?.(theme);
+      if (colors?.length) {
+        const finalColors = colors.map((color) => {
+          return theme.visualization.getColorByName(color);
+        });
```

Comment:
this map should not be needed, handled internally by the color mode

Reply:
This goes directly into a css backgroud property so not sure where it would be handled? Some of the gradients return colors like `panel-bg` which would create `linear-gradient(panel-bg)`. Also `getColors` returns `string[]`.

Reply:
Think with the change I merged yesterday you should get final mapped (real) colors

Reply:
as you can see here: the color mode getColors maps the mode colors to final real colors and caches the result https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/field/fieldColor.ts#L162